syntax = "proto3";

option go_package = "data/lib";

import "types.proto";

enum Window {
  NULL_WINDOW     = 0;
  HOUR            = 1;
  DAY             = 2;
  WEEK            = 3;
  MONTH           = 4;
  QUARTER         = 5;
  YEAR            = 6;
  FOREVER         = 7;
}

message ProtoGetCountRequest {
  CounterType CounterType = 1;
  Window Window = 2;
  repeated uint64 Key = 3;
  uint64 Timestamp = 4;
}

message ProtoGetRateRequest {
  CounterType NumCounterType = 1;
  CounterType DenCounterType = 2;
  repeated uint64 NumKey = 3;
  repeated uint64 DenKey = 4;
  Window Window = 5;
  uint64 Timestamp = 6;
  bool LowerBound = 7;
}
